### PR TITLE
Add org fans summary interface

### PR DESCRIPTION
**Pull request by mikethetike**

### CODE REVIEW

File: src/resources/namespaced/organization-fans.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	/**
+	 * List fans in the organization
+	 * @params {..[[PagingInterface]], organization_id:uuid. q: string, sort?: string [FirstName, LastName, Email, Phone, Orders, FirstOrder, LastOrder, Revenue] }
+	 * @required {organization_id:uuid}
+	 * @data Array<[[OrganizationFanInterface]]>
+	 */
+	index(): IndexInterface {
+		return createRequestMethod({
+			name: "index",
```
Comment (Krakaw) [inline src/resources/namespaced/organization-fans.ts]: `name: "index"` isn't required, it is used as an alias.

File: src/resources/namespaced/organization-fans.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			method: "GET",
+			path: "/{organization_id}/fans?query={q}&page={page}&limit={limit}&sort={sort}&dir={dir}",
```
Comment (Krakaw) [inline src/resources/namespaced/organization-fans.ts]: If you add these items to the url they will be required for every request. 
To add default values use a pipe
`path: "/{organization_id}/fans?query={q|}&page={page|0}&limit={limit|25}&sort={sort|}&dir={dir|asc}",`

File: src/resources/namespaced/organization-fans.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		super("organizations");
+	}
+
+
+	
+
+
+	/**
+	 * List fans in the organization
+	 * @params {..[[PagingInterface]], organization_id:uuid. q: string, sort?: string [FirstName, LastName, Email, Phone, Orders, FirstOrder, LastOrder, Revenue] }
```
Comment (Krakaw) [inline src/resources/namespaced/organization-fans.ts]: ...[[PagingInterface]]

File: src/interfaces/resources/organization-fan.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	first_name?: string;
+	last_name?: string;
+	email?: string;
+	phone?: string;
+	order_count: number;
+	first_order_time: Date;
+	last_order_time: Date;
+	revenue_in_cents: number;
+}
+
```
Comment (Krakaw) [inline src/interfaces/resources/organization-fan.ts]: Add
```suggestion
export const createOrganizationFan = (base: any = {}): OrganizationFanInterface => {
	return {
		...{
			organization_id:"",
	user_id:"",
	first_name: "",
	last_name: "",
	email: "",
	phone: "",
	order_count: "",
	first_order_time: "",
	last_order_time: "",
	revenue_in_cents: 0,
		},
		...base

	}as OrganizationFanInterface;
};
```

Comment (Krakaw) [inline src/interfaces/resources/organization-fan.ts]: Then add the createOrganizationFan to [src/interfaces/resources/index.ts](https://github.com/big-neon/bn-api-node/blob/master/src/interfaces/resources/index.ts#L36)

File: src/resources/namespaced/organization-fans.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+
+import { createRequestMethod } from "../../interfaces/server/request-method.interface";
+import ResourceClass from "../../classes/abstracts/resource.class";
+import { OrganizationInterface } from "../../interfaces/resources/organization.interface";
+import { IndexInterface } from "../../interfaces/resources/structures/index.interface";
+
+/**
+ * @endpoint organizations.fans
+ */
+class OrganizationFansResource extends ResourceClass {
```
Comment (Krakaw) [inline src/resources/namespaced/organization-fans.ts]: Add an `@endpoint organizations.fans [[OrganizationFansResource]]` to [src/resources/index.ts
](https://github.com/big-neon/bn-api-node/blob/master/src/resources/index.ts#L42)

Comment (Krakaw) [inline src/resources/namespaced/organization-fans.ts]: Add `fans: OrganizationFansResource,` to [src/resources/organization.ts](https://github.com/big-neon/bn-api-node/blob/master/src/resources/organization.ts#L21)

File: src/resources/namespaced/organization-fans.ts (hunk 3 above)
Comment (mikethetike) [reply src/resources/namespaced/organization-fans.ts]: Not sure what this cryptic clue means

Comment (Krakaw) [reply src/resources/namespaced/organization-fans.ts]: Sorry, I meant
```suggestion
* @params {...[[PagingInterface]], organization_id:uuid. q: string, sort?: string [FirstName, LastName, Email, Phone, Orders, FirstOrder, LastOrder, Revenue] }
```

**Pull request merged**